evm/v0.5/testv2/VRF/VRF.test.ts

```diff
@@ -0,0 +1,208 @@
+      bn(result[0])
+        .pow(big3)
+        .add(bn(7))
+        .umod(f.fieldSize),
+      bn(result[1])
+        .pow(big2)
+        .umod(f.fieldSize),
+      'y^2=x^3+7',
+    )
+    // See golang code
+    result = await VRF.hashToCurve(f.generator, 1)
+    assertBigNum(
+      result[0],
+      '0x530fddd863609aa12030a07c5fdb323bb392a88343cea123b7f074883d2654c4',
+      'mismatch with output from services/vrf/vrf_test.go/TestVRF_HashToCurve',
+    )
+    assertBigNum(
+      result[1],
+      '0x6fd4ee394bf2a3de542c0e5f3c86fc8f75b278a017701a59d69bdf5134dd6b70',
+      'mismatch with output from services/vrf/vrf_test.go/TestVRF_HashToCurve',
+    )
+  })
+
+  it('Correctly verifies linear combinations with generator', async () => {
+    assert(
+      await VRF.verifyLinearCombinationWithGenerator(
+        5,
+        f.twiceGenerator,
+        7,
+        h.pubkeyToAddress(f.seventeenTimesGenerator),
+      ),
+      '5*(2*g)+7*g=17*g?',
+    )
+  })
+
+  it('Correctly computes full linear combinations', async () => {
+    const projSum = await VRF.projectiveECAdd(
+      f.eightTimesGenerator[0],
+      f.eightTimesGenerator[1],
+      f.nineTimesGenerator[0],
+      f.nineTimesGenerator[1],
+    )
+    const zInv = projSum[2].invm(f.fieldSize)
+    assertPointsEqual(
+      f.seventeenTimesGenerator,
+
+      // '4*(2*g)+3*(3*g)=17*g?'
+      await VRF.linearCombination(
+        4,
+        f.twiceGenerator,
+        f.eightTimesGenerator,
+        3,
+        f.thriceGenerator,
+        f.nineTimesGenerator,
+        zInv,
+      ),
+    )
+  })
+
+  it('Computes the same hashed scalar from curve points as the golang code', async () => {
+    const scalar = await VRF.scalarFromCurve(
+      f.generator,
+      f.generator,
+      f.generator,
+      h.pubkeyToAddress(f.generator),
+      f.generator,
+    )
+    assertBigNum(
+      '0x2b1049accb1596a24517f96761b22600a690ee5c6b6cadae3fa522e7d95ba338',
+      scalar,
+      'mismatch with output from services/vrf/vrf_test.go/TestVRF_ScalarFromCurve',
+    )
+  })
+
+  it('Knows a good VRF proof from bad', async () => {
+    const x = big1 // "secret" key in Goldberg's notation
+    const pk = f.generator
+    const seed = 1
+    const hash = await VRF.hashToCurve(pk, seed)
+    const gamma = hash // Since gamma = x * hash = hash
+    const k = big1 // "Random" nonce, ha ha
+    const u = f.generator // Since u = k * generator = generator
+    const v = hash // Since v = k * hash = hash
+    const c = await VRF.scalarFromCurve(
+      hash,
+      pk,
+      gamma,
+      h.pubkeyToAddress(u),
+      v,
+    )
+    const s = k.sub(c.mul(x)).umod(f.groupOrder) // s = k - c * x mod group size
+    const cGamma = [
+      // >>> print("'0x%x',\n'0x%x'" % tuple(s.multiply(gamma, c)))
+      bn('0xa2e03a05b089db7b79cd0f6655d6af3e2d06bd0129f87f9f2155612b4e2a41d8'),
+      bn('0xa1dadcabf900bdfb6484e9a4390bffa6ccd666a565a991f061faf868cc9fce8'),
+    ]
+    const sHash = [
```

Comment:
should `sHash` and `cGamma` be moved to `VRF/fixtures.ts`?

Reply:
I didn't move them over yet since they're scoped to that specific test, I don't have enough context of the test to move it out and lose the scope it's under